autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/AutofillJavascriptInterface.kt

```diff
@@ -169,12 +203,61 @@ class AutofillStoredBackJavascriptInterface @Inject constructor(
             val jsCredentials = JavascriptCredentials(request.credentials!!.username, request.credentials.password)
             val credentials = jsCredentials.asLoginCredentials(currentUrl)
 
-            withContext(dispatcherProvider.main()) {
-                callback?.onCredentialsAvailableToSave(currentUrl, credentials)
+            val autologinId = autoSavedLoginsMonitor?.getAutoSavedLoginId(tabId)
+            Timber.i("Autogenerated? %s, Previous autostored login ID: %s", request.credentials.autogenerated, autologinId)
+            val autosavedLogin = autologinId?.let { autofillStore.getCredentialsWithId(it) }
+
+            val actions = passwordEventResolver.decideActions(autosavedLogin, request.credentials.autogenerated)
+            processStoreFormDataActions(actions, currentUrl, credentials, autologinId)
+        }
+    }
+
+    private suspend fun processStoreFormDataActions(
+        actions: List<Actions>,
+        currentUrl: String,
+        credentials: LoginCredentials,
+        autologinId: Long?,
+    ) {
+        Timber.d("%d actions to take: %s", actions.size, actions.joinToString())
+        actions.forEach {
+            when (it) {
+                is DeleteAutoLogin -> {
+                    autofillStore.deleteCredentials(it.autologinId)
+                }
+
+                DiscardAutoLoginId -> {
```

Comment:
Not strictly needed because `DiscardAutoLoginId` is an `object`, but it flows better if they all have use `is` and I don't think there's a downside. Will add it in for symmetry with the others that are not `object`s.